pkg/policy/rule.go

```diff
@@ -410,6 +427,15 @@ func (r *rule) canReachIngress(ctx *SearchContext, state *traceState) api.Decisi
 	}
 
 	state.selectRule(ctx, r)
+
+	// If a requirement explicitly denies access, we can return.
+	if r.meetsRequirementsIngress(ctx, state) == api.Denied {
+		return api.Denied
+	}
+
+	// At this point we only know whether anything is not denied by requirements.
+	// Now we need to determine whether rule allows traffic based off of
+	// SearchContext.
```

Comment:
This comment is fine, but AFAICT the code immediately below was already run inside `r.meetsRequirementsIngress()`. I think the comment is trying to describe what the code below this block will do in `return r.canReachFromEndpoints()`. In other words, the iteration block immediately below this comment that ranges over the `r.FromRequires` in each `r.Ingress()` is duplicate and can be removed.

Reply:
Good catch - I think my brain remembered to delete it on the egress side, and thought it had done it on the ingress side too. Boilerplate =(